frontend/src/domains/utils/createCategoryList.ts

```diff
@@ -0,0 +1,20 @@
+import { CATEGORY_LIST, CategoryListType } from '@/constants/categoryList';
+import { SelectedCategoryItem } from '../admin/CreateRoomModal/hooks/useCategorySelection';
+
+export function createCategoryIconPairs(initialCategories: CategoryListType[]) {
+  return initialCategories.reduce(
+    (acc: SelectedCategoryItem[], categoryString) => {
+      const categoryItem = CATEGORY_LIST.find(
+        (item) => item.category === categoryString
+      );
+      if (categoryItem) {
+        acc.push({
+          icon: categoryItem.icon,
+          category: categoryItem.category,
+        });
+      }
+      return acc;
+    },
+    []
+  ) as SelectedCategoryItem[];
```

Comment:
```suggestion
 return initialCategories
    .map((categoryString) =>
      CATEGORY_LIST.find((item) => item.category === categoryString)
    )
    .filter((item): item is SelectedCategoryItem => !!item)
    .map((item) => ({
      icon: item.icon,
      category: item.category,
    }));
```

요렇게 filter랑 map을 이용해서 나타내면 더 깔끔하게 나타낼 수 있을 것 같아요 :)
물룐 필수는 아님